code/modules/clothing/glasses/glasses.dm

```diff
@@ -26,28 +26,40 @@
 
 /obj/item/clothing/glasses/attack_self(mob/user)
 	if(toggleable)
-		if(active)
-			active = 0
-			icon_state = deactive_state
-			user.update_inv_glasses()
-			to_chat(user, "You deactivate the optical matrix on [src].")
-			playsound(user, 'sound/items/googles_off.ogg', 15)
-		else
-			active = 1
-			icon_state = initial(icon_state)
-			user.update_inv_glasses()
-			to_chat(user, "You activate the optical matrix on [src].")
-			playsound(user, 'sound/items/googles_on.ogg', 15)
-
-		if(ishuman(loc))
-			var/mob/living/carbon/human/H = loc
-			if(H.glasses == src)
-				H.update_tint()
-				H.update_sight()
-
-		update_action_button_icons()
+		toggle_glasses(user)
 
 
+/obj/item/clothing/glasses/proc/toggle_glasses(mob/user)
+	if(active)
+		deactivate_glasses(user, TRUE)
+	else
+		activate_glasses(user, TRUE)
+	if(ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		if(H.glasses == src)
+			H.update_tint()
+			H.update_sight()
+
+	update_action_button_icons()
+
+
+/obj/item/clothing/glasses/proc/activate_glasses(mob/user, feedback = FALSE)
```

Comment:
Feedback is typically referenced as `silent` elswhere, but i dont mind either way.

Reply:
Change request done.